pennylane/utils.py

```diff
@@ -85,7 +85,7 @@ def decompose_hamiltonian(H, hide_identity=False):
     coeffs = []
 
     for term in itertools.product(paulis, repeat=n):
-        matrices = [i._matrix() for i in term]
+        matrices = [i.compute_matrix() for i in term]
```

Comment:
This assumes that all terms in the hamiltonian are ops that do not take operators!